File: guides/micronaut-dependency-injection-types/kotlin/src/main/kotlin/example/micronaut/methodparameter/MessageController.kt
package example.micronaut.methodparameter

import example.micronaut.MessageService
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import jakarta.inject.Inject

@Controller("/setter")
class MessageController {
    var messageService: MessageService? = null

    @Get
    @Produces(MediaType.TEXT_PLAIN)
    fun index() = messageService?.compose()

    @Inject
    fun populateMessageService(messageService: MessageService) {
        this.messageService = messageService
    }
}

File: guides/micronaut-dependency-injection-types/python/src/example/micronaut/methodparameter/message_controller.py
from jakarta.inject import Inject
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get

from example.micronaut.message_service import MessageService


@Controller("/setter")
class MessageController:
    def __init__(self):
        self.message_service: MessageService | None = None

    @Get(produces=MediaType.TEXT_PLAIN)
    def index(self) -> str:
        assert self.message_service is not None
        return self.message_service.compose()

    @Inject
    def populate_message_service(self, message_service: MessageService):
        self.message_service = message_service
